examples/digital-asset-manager/src/components/Assets.tsx

```diff
@@ -24,70 +24,72 @@ const Assets = ({
   const [assetUrls, setAssetUrls] = useState<ImageData[]>([]);
   const [assetCount, setAssetCount] = useState(0);
 
-  const fetchAssetsByCollectionId = () => {
-    fetch(`/.netlify/functions/collectionId`, {
-      method: "POST",
-      body: JSON.stringify({ collectionId }),
-    })
-      .then((response) => {
-        return response.json();
-      })
-      .then((result) => {
-        setAssetUrls([...result]);
-        setAssetCount(result.length);
-        setLoading(false);
+  const fetchAssetsByCollectionId = async () => {
+    try {
+      const collectionData = await fetch(`/.netlify/functions/collectionId`, {
+        method: "POST",
+        body: JSON.stringify({ collectionId }),
       });
+
+      const result = await collectionData.json();
+
+      setAssetUrls([...result]);
+      setAssetCount(result.length);
+      setLoading(false);
+    } catch (error) {
+      console.error(error);
+    }
   };
 
   useEffect(() => {
     fetchAssetsByCollectionId();
```

Comment:
Setting the results into state is a side effect. Return the results and put them into state from within `useEffect`

Reply:
You called your function `fetchAssetsByCollectionId`. As a reader, I'd expect that to return something, not change anything in state. So a solution would be to rename the function to `fetchAssetsByCollectionIdAndSetThemInState`, which is a bit of a mess. Separation of concerns would be better here:


```ts
useEffect(() => {
  const getData = async () => {
  	const assets = await fetchAssetsByCollectionId();
    setThemInstate(assets);
  }
 
  getData();
  
})
```